packages/context/test/acceptance/extension-point.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Context, Binding, BindingScope, inject} from '../..';
+
+// tslint:disable:no-any
+/**
+ * Interface for the extension point configuration
+ */
+export interface ExtensionPointConfig {
+  extensions: {
+    [extensionName: string]: any;
+  };
+  extensionPoint: {
+    [property: string]: any;
+  };
+}
+
+/**
+ * Base class for extension points
+ */
+export abstract class ExtensionPoint<EXT extends object> {
+  /**
+   * Configuration (typically to be injected)
+   */
+  protected configuration: ExtensionPointConfig;
+
+  constructor(
+    /**
+     * The unique name of this extension point. It also serves as the binding
+     * key prefix for bound extensions
+     */
+    protected name: string,
+    /**
+     * The Context (typically to be injected)
+     */
+    protected context: Context,
+    /**
+     * Configuration (typically to be injected)
+     */
+    configuration?: ExtensionPointConfig,
+  ) {
+    this.configuration = configuration || {extensionPoint: {}, extensions: {}};
+  }
+
+  /**
+   * Find an array of bindings for extensions
+   */
+  getExtensionBindings(): Binding[] {
+    return this.context.find(`${this.name}.*`);
+  }
+
+  /**
+   * Get a map of extension bindings by the keys
+   */
+  getExtensionBindingMap(): {[name: string]: Binding} {
+    const extensions: {[name: string]: Binding} = {};
+    const bindings = this.context.find(`${this.name}.*`);
+    bindings.forEach(binding => {
+      extensions[binding.key] = binding;
+    });
+    return extensions;
+  }
+
+  /**
+   * Look up an extension binding by name
+   * @param name Name of the extension
+   */
+  getExtensionBinding(name: string): Binding {
+    const key = `${this.name}.${name}`;
+    const extensions = this.getExtensionBindingMap();
+    return extensions[key];
+  }
+
+  /**
+   * Get an instance of an extension by name
+   * @param name Name of the extension
+   */
+  async getExtension(name: string): Promise<EXT> {
+    const binding = this.getExtensionBinding(name);
+    // Create a child context to bind `config`
+    const extensionContext = new Context(this.context);
```

Comment:
We need to carefully consider implications for CONTEXT-scoped bindings, as they will be cached at the level of this temporary extension context. Is this the right thing to do? Shouldn't we cache CONTEXT-scoped bindings at `this.context` level instead?

Reply:
Let me think about that.

Meanwhile, I'll evaluate the possibility to make `ExtensionPoint` extend `Context`. This way, each extension point will have its own isolated context.